_docs/automation-time-apis.md

```diff
@@ -121,6 +121,180 @@ pub enum Error {
 }
 ```
 
+### RPC: Generate Task ID based on Provided ID and Account ID
+This API allows you to generate the Task ID based on the Provided ID and Account ID used to create the task.
+
+#### Call
+```rust
+fn generate_task_id(
+    /// The `account_id` of the user. This is the wallet address of the user who created the task.
+    account_id: AccountId,
+    /// An id provided by the user. This id must be unique for each task for a given user.
+    provided_id: Vec<u8>,
+    )
+```
+
+** For those who are not familiar with rust `Vec<u8>` accepts a string input. For example, "I am as unique as a snowflake" is a valid input.
+** This RPC should return the task ID associated with this unique combination of account ID and provided ID.
+** This RPC does not provide the task ID for only tasks that are scheduled. It can provided the potential task ID for non-scheduled tasks, so long as the non-scheduled tasks are scheduled with the provided account_id and provided_id. 
+
+#### Errors
+```rust
+pub enum Error {
+    // Could not generate the task_id.
+    "Unable to generate task_id",
```

Comment:
Putting this string in a rust enum looks strange to me. For folks that don't know rust the `enum` won't help and it will confuse devs that are familiar with rust. Maybe just a bulleted list of these strings?